shared-module/audiocore/__init__.h

```diff
@@ -31,13 +31,36 @@
 #include <stdint.h>
 
 #include "py/obj.h"
+#include "py/proto.h"
 
 typedef enum {
     GET_BUFFER_DONE,            // No more data to read
     GET_BUFFER_MORE_DATA,       // More data to read.
     GET_BUFFER_ERROR,           // Error while reading data.
 } audioio_get_buffer_result_t;
 
+typedef uint32_t (*audiosample_sample_rate_fun)(void* sample_obj);
+typedef uint8_t (*audiosample_bits_per_sample_fun)(void* sample_obj);
+typedef uint8_t (*audiosample_channel_count_fun)(void* sample_obj);
+typedef void (*audiosample_reset_buffer_fun)(void* sample_obj);
+typedef audioio_get_buffer_result_t (*audiosample_get_buffer_fun)(void* sample_obj,
+        bool single_channel, uint8_t channel, uint8_t** buffer,
+        uint32_t* buffer_length);
+typedef void (*audiosample_get_buffer_structure_fun)(void* sample_obj,
```

Comment:
These sample_objs can also be mp_obj_t right? That'd make it clearer that the first field is the Python type pointer.